From 126bb51ea237d0a8cfe5a0b1397d02932b3131b8 Mon Sep 17 00:00:00 2001
From: "Jan Alexander Steffens (heftig)" <heftig@archlinux.org>
Date: Wed, 28 Oct 2026 04:24:43 +0100
Subject: [PATCH] tests: Add missing check for Xvfb

This test, introduced in v51.2, failed with a crash in GTK when Xvfb
is not available.
---
 tests/ui/test-panel-history.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/tests/ui/test-panel-history.c b/tests/ui/test-panel-history.c
index b5e87826e..fdc93d5c1 100644
--- a/tests/ui/test-panel-history.c
+++ b/tests/ui/test-panel-history.c
@@ -214,6 +214,12 @@ test_delete_key (void)
 static void
 test_drag_activates_nothing (void)
 {
+    if (!have_display)
+    {
+        g_test_skip ("A private Xvfb display is required");
+        return;
+    }
+
     g_autoptr (GError) error = NULL;
     g_autoptr (GPasteClient) client = g_paste_client_new_sync (&error);
 
